code/payments/Tally-Vs-Rollup.md

```diff
@@ -0,0 +1,69 @@
+# Tally vs Rollup
+
+The terms "tally" and "rollup" are used throughout the code base. Their specific meanings have been a source of confusion since their meanings are similar.
+
+The purpose of this document is to:
+- define the terms "tally" and "rollup".
+- provide details about how tally and rollup are used in the code base.
+
+## Definitions
+
+Both tally and rollup sum an amount of something (i.e. storage or bandwidth) as it relates to a period of time.
+
+They differ in what the period of time is.
+- For tally, the period of time is just a single moment in time, typically the most recent or current moment.
+- For rollup, the period of time is a time frame, for example 1 hr or 1 day time interval.
+
+This means that tally is a snapshot of the current state. And rollup is an aggregate of those snapshots.
```

Comment:
* Tally is an eventually consistent sum of values, where the computation ended at a specific moment in time.
* Rollup is an sum of values that happened over an interval in time.

Reply:
this sentence is confusing: "Tally is an eventually consistent sum of values, where the computation ended at a specific moment in time.".

Is there a simpler way of saying that?

Reply:
* "Tally sums over the values incrementally, possibly missing some values that are added during tallying."
* "Tally sums over the values incrementally. Since tallying is a slow process it may miss data that has been inserted or deleted after tallying was started."

Reply:
* "However they will be caught on the next run."

Reply:
I think Egon is trying to point out that the tally service is not atomic. The tally service may end up coming up with a sum that was never actually a moment-in-time total. 

For instance, let's say there are two 1MB objects in the system, /my/file/a, and /my/file/b. Tally runs and looks at /my/file/a, which is 1MB, so it adds 1MB to the total. It ends with /my/file/a. Right before it gets to /my/file/b, /my/file/a is updated and changes to 1.5MB, and then /my/file/b is updated and changes to .5MB. Tally will result in a total of 1.5MB, which was never the total (it was 2, then 2.5, then 2).

So, if tally keeps running past the point where files stop changing, then eventually, tally and the files will be consistent.

I agree there isn't a great way of saying this that is better than "eventually consistent", other than that tally is operating on changing data and not atomically operating on a snapshot. 

Does that make sense?

Reply:
That all makes sense.  However, I'm just trying to come up with 1 sentence that defines "Tally" in a simple way that people can understand when they read it.   

A one sentence definition will cut out many details and nuances.  It is possible that we cannot define "tally" in one simple sentence, but I would like to try.

How would you define tally in 1 (or 2) simple sentence?

Reply:
perhaps: "A tally is a non-atomic summed total of some column of all of the entries in a database"?

Reply:
"Tally is a sum of values that is evaluated over a time, where any event that happens during tallying may or may not be included in the sum."

Reply:
ok thank you both. i will make these changes.